e2e/stress/RaceConditionTest/Program.cs

```diff
@@ -0,0 +1,141 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+// Run using the following:
+//  dotnet run --framework netcoreapp2.1 --configuration Release
+//  dotnet run --framework net47 --configuration Release
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Azure.Devices.Client;
+using System.Diagnostics;
+using System.Net.Sockets;
+using Microsoft.Azure.Devices.Client.Exceptions;
+using System.Threading;
+using System.Text;
+using System.IO;
+
+namespace Microsoft.Azure.Devices.E2ETests
+{
+    public class MessageRaceConditionTest
+    {
+        private string DeviceConnectionString;
+        private const int NumberOfParallelDeviceClientCalls = 10000;
+
+        public MessageRaceConditionTest(string deviceConnectionString)
+        {
+            DeviceConnectionString = deviceConnectionString;
+        }
+
+        public async Task<int> RunTest_SendTenThousandMessagesAsync(TransportType transportType)
+        {
+            const int MaxExecutionTimeMinutes = 10;
+            DeviceClient deviceClient = DeviceClient.CreateFromConnectionString(DeviceConnectionString, transportType);
+            deviceClient.OperationTimeoutInMilliseconds = 5;
+
+            CancellationTokenSource cancellationTokenSource = new CancellationTokenSource(TimeSpan.FromMinutes(MaxExecutionTimeMinutes));
+
+#pragma warning disable CS0618 // Type or member is obsolete
+            deviceClient.RetryPolicy = RetryPolicyType.No_Retry;
```

Comment:
This isn't the right way to disable RetryPolicy. You need to use `SetRetryPolicy(IRetryPolicy)` instead.
AFAIK, this line is a no-op.